server/src/main/java/org/elasticsearch/index/codec/tsdb/ES87TSDBDocValuesProducer.java

```diff
@@ -1106,8 +1106,8 @@ long advance(long index) throws IOException {
                 final long blockIndex = index >>> ES87TSDBDocValuesFormat.NUMERIC_BLOCK_SHIFT;
                 final int blockInIndex = (int) (index & ES87TSDBDocValuesFormat.NUMERIC_BLOCK_MASK);
                 if (blockIndex != currentBlockIndex) {
-                    assert blockIndex > currentBlockIndex;
-                    if (blockIndex - 1 > currentBlockIndex) {
```

Comment:
I see this exact pattern in `NumericDocValues#longValue`. Do we have off-by-one errors there as well?

https://github.com/elastic/elasticsearch/blob/ae69a61fdb326e69e422cc4e055622d9c52bcf36/server/src/main/java/org/elasticsearch/index/codec/tsdb/ES87TSDBDocValuesProducer.java#L1015

https://github.com/elastic/elasticsearch/blob/ae69a61fdb326e69e422cc4e055622d9c52bcf36/server/src/main/java/org/elasticsearch/index/codec/tsdb/ES87TSDBDocValuesProducer.java#L1075

?

Reply:
Thanks Bent. These are single value paths, so they shouldn't be affected by this bug. Anyway, I've extended the tests and made the seeking patterns more consistent in this class.